### PR TITLE
Modified Image URL Generator (Part A)

### PR DESCRIPTION
**Pull request by Temidayo32**
This PR addresses Part A as part of the implementation listed on issue #3683 by @lb- . 

_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by Temidayo32** [comment]
@lb- Please review

**Comment by lb-** [comment]
Very cool. Thank you. 

Would you please be able to add some before/after screenshots - maybe with a small generated image and a large one for each. 

This way we can get some quick feedback hopefully from other team members. 

I haven't done a full review but just thought it would be good to avoid styling against loading-mask. Maybe a more suitable class name added such as preview-container. Nitpicky but just makes it easier to understand that it's not specifically related to the loading behaviour.

We may need to move the CSS to a scss file or see if we can do this all with tailwind classes.  But that can be revisited once a proper UI review has been done 

Thanks again for getting this going!

**Comment by Temidayo32** [comment]
Yes, I would do that and have it sent. 

> Would you please be able to add some before/after screenshots - maybe with a small generated image and a large one for each.

But I am little confused what you mean here. Can you clarify further? Thanks

**Comment by Temidayo32** [comment]
@lb-  Here is are before and after pictures

Large Image (before)
screencapture-localhost-8000-admin-images-53-generate-url-2023-11-04-10_58_27

 After
 
screencapture-localhost-8000-admin-images-54-generate-url-2023-11-04-10_33_06

Small Image (before)
screencapture-localhost-8000-admin-images-54-generate-url-2023-11-04-10_55_15

After

screencapture-localhost-8000-admin-images-53-generate-url-2023-11-04-10_52_10

File: wagtail/images/templates/wagtailimages/images/url_generator.html [hunk 1]
```diff
@@ -17,20 +29,20 @@
...
-        {% panel id="url" icon="link" heading=heading %}
-            <textarea id="result-url" rows="1"></textarea>
-        {% endpanel %}
-
         {% trans "Preview" as heading %}
         {% panel id="preview" icon="view" heading=heading %}
             <div>
-                <div class="loading-mask inline-block">
-                    <img class="preview" src="" alt="{% trans 'Preview' %}" />
+                <div class="loading-mask">
```
Comment (lb-) [inline wagtail/images/templates/wagtailimages/images/url_generator.html:35]:
```suggestion
                <div class="loading-mask w-block w-border-2 w-border-border-furniture w-overflow-scroll w-p-4 w-rounded-sm">
```

We can get pretty far here with Tailwind, we cannot do the max-height item for now but the rest can be done this way.

**Review by lb- — APPROVED** [review]
Thank you @Temidayo32  - this is looking great.

I have opted for a bit more Tailwind usage, we may want to avoid the style block in a future release (CSP compliance) but for now it's still required for the max-height usage.

I have made these classes changes and will force push & merge, I also included a smaller padding in the preview + border radius, it felt a bit too large.

I have adjusted some classes on this PR also, just to align with other `w-prefix` BEM class usage.

Screenshot 2023-11-09 at 7 42 07 am

Thank you for looking into this, this feels so much better and it's less buggy, less reliant on JavaScript!. Thanks to @thibaudcolas for the suggested approach many moons ago.

**Comment by Temidayo32** [comment]
@lb- thanks so much for reviewing thus

**Pull request merged**